gen/__init__.py

```diff
@@ -306,20 +305,67 @@ def do_gen_package(config, package_filename):
     log.info("Package filename: %s", package_filename)
 
 
-def extract_files_with_path(start_files, paths):
+def render_late_content(content, late_values):
+    # TODO(branden): Unit tests
+
+    def _next_substring(string, substring, start=0):
+        idx = string.find(substring, start)
+        if idx < 0:
+            return None
+        return idx, idx + len(substring)
+
+    def _interpolate_values(string):
+        if not string:
+            return
+
+        # Find the next placeholder.
+        placeholder_start = _next_substring(string, gen.internals.LATE_BIND_PLACEHOLDER_START)
+        if not placeholder_start:
+            # No placeholder found in the string.
+            yield string
+            return
+
+        # Yield the string preceding the placeholder, if any.
+        if placeholder_start[0] > 0:
+            yield string[:placeholder_start[0]]
+
+        # Find the end of the placeholder.
+        placeholder_end = _next_substring(string, gen.internals.LATE_BIND_PLACEHOLDER_END, placeholder_start[1])
+        if not placeholder_end:
+            raise Exception("Bad late config file: Can't find end of placeholder")
+
+        # Extract the name between the start and end and yield its value.
+        name = string[placeholder_start[1]:placeholder_end[0]]
+        if name not in late_values:
+            raise Exception('Bad late config file: Found placeholder for unknown value "{}"'.format(name))
+        yield late_values[name]
+
+        # Interpolate the rest of the string following the placeholder.
+        yield from _interpolate_values(string[placeholder_end[1]:])
+
+    return ''.join(_interpolate_values(content))
+
+
+def resolve_late_package(config, late_values):
+
+    return {
+        'package': [
+            {k: render_late_content(v, late_values) if k == 'content' else v for k, v in fileinfo.items()}
+            for fileinfo in config['package']
+        ]
+    }
+
+
+def extract_files_containing_late_variables(start_files):
     found_files = []
-    found_file_paths = []
     left_files = []
 
```

Comment:
nit / bug from me: Would be nice to check LATE_BLIND_PLACEHOLD_START never is in any other key or value of the file_info dictionary and error early here. I forsee someone trying to late bind part of a path and it breaking badly.